web/src/compose_validate.js

```diff
@@ -250,6 +254,9 @@ function show_wildcard_warnings(stream_id) {
             wildcard_template,
             $("#compose_banners"),
         );
+    } else {
+        // if there is already a banner, replace it with the new one
+        $(`#compose_banners .${CSS.escape(classname)}`).replaceWith(wildcard_template);
```

Comment:
Let's replace this with a `compose_banner` method to avoid too much of an abstraction boundary violation. Also see 6dc10f8696f432b47ae2e96ad9715d0ae3e94c71 and https://github.com/zulip/zulip/pull/25422; we should make sure we don't make it harder to have this code be correct for the message editing code path.

Reply:
@timabbott So I abstracted that logic in a method `update_or_append_banner` is that ok and I think #25422 will probably be fine after merging this except some merge conflicts of course.